delocate/delocating.py

```diff
@@ -578,6 +592,274 @@ def _make_install_name_ids_unique(
         validate_signature(lib)
 
 
+def _get_macos_min_version(dylib_path: Path) -> Iterator[tuple[str, Version]]:
+    """Get the minimum macOS version from a dylib file.
+
+    Parameters
+    ----------
+    dylib_path : Path
+        The path to the dylib file.
+
+    Yields
+    ------
+    str
+        The CPU type.
+    Version
+        The minimum macOS version.
+    """
+    if dylib_path.is_dir() or not _is_macho_file(dylib_path):
```

Comment:
Does `_is_macho_file` break if the path is a directory? If that's the case then the function should be updated.
```suggestion
    if not _is_macho_file(dylib_path):
```